### ISSUE TITLE
 OnOpenPage seems to be firing before sorted and filtering of the table.

### ISSUE THREAD

**Issue by Update-Mark** [microsoft/AL#5338]
**Describe the bug**
I have a client who would simply like to reverse the sort order of the Sales Order List in BC14. They know that they can click on the column, but it's a convenience issue for them.

So created a page extension and added the following:
```
trigger OnOpenPage()
 begin
        rec.SetCurrentKey("Document Type", "No.");
        rec.SetAscending("No.", false);
 end;
```
and published and ran the Sales Order List....nothing.  
Ran it in the Windows Client...works.
I know this used to be possible, and I am thinking that the web client might be processing the trigger slightly differently then the windows client. 
Has anyone gotten this to work correctly in the web client?
Note;Also wired up to the onopenpage event, No luck there either.

**To Reproduce**
Steps and to reproduce the behavior:

1. Create an extension that modifies the sorting key on a base page object.
2. Add sorting code on the OnOpenPage Trigger

``` AL Code to reproduce the issue ```
```
trigger OnOpenPage()
 begin
        rec.SetCurrentKey("Document Type", "No.");
        rec.SetAscending("No.", false);
 end;
```
**Expected behavior**
List should sort descending by No.

**5. Versions:**

 - AL Language: 3.0.106655

 - Business Central: 14.0.32600.0

**Comment by update-jeremy** [comment]
I can confirm this is a sequence issue.  In the Classic / RTC environment, during the OnOpenPage trigger, you can change your FILTERGROUPs to 3 or 4 to get the link/view.  In the BC Web Environments, OnOpenPage, even if it's a factbox linked from a Sales Order, if you cycle through FILTERGROUPS 1-10, they're all 100% empty.

For a User Story on this, this means that Extensions can't hook up to the OnOpenPage event to apply changes to filters and sorting (as per Mark's case above).  This is particularly unhelpful if you're creating a new custom List Page that will work as a Lookup page, for example.

This also prevents generation of Temporary Record sets during OnOpenPage from supplied record Filters.  

You cannot save, react, or read the filters that were present during OnOpenPage.

**Comment by thpeder** [comment]
Let's get the issue to a team who will fix and backport it, as this repository is only for issues related to the AL compiler in latest developer preview environment for Dynamics 365 Business Central. We suggest that you open a support case or file a bug in Collaborate, to ensure that all Business Central users benefit from your catch as soon as possible.

To open a support case, you can:

- Open a support request to CSS through the PartnerSource portal
- Contact your Service Account Manager (SAM) in your local subsidiary to understand what is included in your contract as of support incident and PAH (Partner Advisory Hours). Your SAM might also direct you "step by step" how to open a support request or how to get credentials, if this is the first time for you or your company

If you file the bug in Collaborate, remember to include steps to reproduce the issue, and the Business Central build number and country version you're using.

**Comment by fvet** [comment]
For reference : encountered the same and reported via collaborate.
https://partner.microsoft.com/en-us/dashboard/collaborate/engagements/1173/feedback/wits/Bugs/273405

**Comment by jehelles** [comment]
We have received a support request referencing this Github topic, and have not been able to reproduce the behavior you see the Windows client (filter group 4 being populated in the OnOpenPage trigger).

Can you please provide the version number of Business Central / NAV where you observed this  behavior, and a small repro (AL extension or txt files).

**Comment by daansaveyn** [comment]
> We have received a support request referencing this Github topic, and have not been able to reproduce the behavior you see in the Windows client (filter group 4 being populated in the OnOpenPage trigger).
> 
> Can you please provide the version number of Business Central / NAV where you observed this behavior, and a small repro (AL extension or txt files).

Please find a sample extension in the attachments.

 A page extension was made on the Customer Statistcis Factbox. This extension contains the following:

- A message in the OnOpenPage trigger displaying the filters in FilterGroup 4. As you can see in the animation below, no filters are found. This while the Customer Statistics Factbox was added via the SubPageLink property:

![image](https://user-images.githubusercontent.com/48217797/76530440-06208580-6474-11ea-8e72-4fbb26809f33.png)

![FilterGroup4](https://user-images.githubusercontent.com/48217797/76530554-28b29e80-6474-11ea-9e1d-6a387ade1702.gif)

- An action "Custom Page" opening a second custom page using the RunPageView property. The custom page contains also a message in the OnOpenPage trigger displaying the filters in FilterGroup 3. The animation below illustrates that in this case filters are found.

![FilterGroup3](https://user-images.githubusercontent.com/48217797/76530739-77603880-6474-11ea-9ade-b4d2bead02ba.gif)

The behavior of the FilterGroup function is not consistent in the OnOpenPage trigger. As a developer it's quite difficult to identify in which triggers an for which filtergroup the functionality can be used.

[SampleFilterGroup4.zip](https://github.com/microsoft/AL/files/4324573/SampleFilterGroup4.zip)

**Comment by jehelles** [comment]
Thank you @daansaveyn for the repro, 

However, to efficiently investigate this bug, I will need a repro that works in the Windows client but not in the web client, _and_ a version number where this behavior is seen.

**Comment by update-jeremy** [comment]
I worked my way backwards through the versions.  You are 100% correct, as it turns out.  I had to go all the way back to Forms in the Classic client to actually get anywhere.  The Windows client has never had access to the FilterGroups from SubPageLink during OnOpenPage.  Attaching a 2009 Text object with Tables/Forms/Pages showing the cases, as well as the results here:

![OnOpenPage-Results](https://user-images.githubusercontent.com/35526546/76843342-ec988880-683b-11ea-872f-767115927ddf.png)

[OnOpenPage_2009_Demo.txt](https://github.com/microsoft/AL/files/4342643/OnOpenPage_2009_Demo.txt)
